ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_pybind.cpp

```diff
@@ -440,9 +440,10 @@ void py_bind_conv2d(py::module& module) {
         &Conv2dConfig::dtype,
         R"doc(Specifies the data type of the output tensor. Supports ttnn.float32, ttnn.bfloat16 and ttnn.bfloat8_b. )doc");
     py_conv_config.def_readwrite("weights_dtype", &Conv2dConfig::weights_dtype, R"doc(
```

Comment:
do we need changes to prepare_conv_weights family of functions in pybind?

Reply:
Do we need some work to make it optional in python to match new C++ api

Reply:
Ok on Conv2d api weights dtype is optional.
In prepare weights dtype is optional for weights.
Why don't we set this to optional in pybind for prepare weights functions?